src/modules/Shared/HOC/privateRoute.tsx

```diff
@@ -18,6 +20,16 @@ const PrivateRoute: React.FC<IProps> = ({ Component }) => {
     return null;
   }
 
+  //check if user has valid role to access the component
+  if (!Role.includes(auth.role)) {
+    if (auth.role === ROLES.EMPLOYEE) {
```

Comment:
Can we have better condition handling here. Like if auth.role isn't present in our ROLE constants it should be always Unauthorized access. If we need some additional condition checking for Employee role we can have a separate if loop with comments which specify the need of checking it